common/strings.c

```diff
@@ -422,42 +422,24 @@ TDNFReplaceString(
     goto cleanup;
 }
 
-uint32_t
-TDNFTrimSuffix(
-    char* pszSource,
-    const char* pszSuffix
-    )
+uint32_t TDNFTrimSuffix(char *pszSource, const char *pszSuffix)
 {
-    uint32_t dwError = 0;
-    int nSourceStrLen = 0, nSuffixStrLen = 0;
+    char *ptr;
 
     if (IsNullOrEmptyString(pszSource) || IsNullOrEmptyString(pszSuffix))
     {
-        dwError = ERROR_TDNF_INVALID_PARAMETER;
-        BAIL_ON_TDNF_ERROR(dwError);
+        return ERROR_TDNF_INVALID_PARAMETER;
     }
 
-    nSourceStrLen = strlen(pszSource);
-    nSuffixStrLen = strlen(pszSuffix);
-
-    if (nSuffixStrLen > nSourceStrLen)
+    ptr = strcasestr(pszSource, pszSuffix);
+    if (!ptr)
     {
-        dwError = ERROR_TDNF_INVALID_PARAMETER;
-        BAIL_ON_TDNF_ERROR(dwError);
+        return ERROR_TDNF_INVALID_PARAMETER;
     }
 
-    while (nSuffixStrLen > 0 &&
```

Comment:
It looks to me that the original function would strip anything from the end that is matching, not only if the whole suffix matches. For example `TDNFTrimSuffix("foobar", "feebar")` would result in `"foo"`, but the new one would return an error.

There is only one caller (in `client/repo.c`), and looks like this is being checked before, so that's probably okay (maybe a bug in the old implementation even).

Reply:
No it doesn't. Newer version of code also return `foo`
Try it here: https://onlinegdb.com/GKQq_joZD

Reply:
For `TDNFTrimSuffix("foobar", "feebar")` it returns an error, I just tried. I modified the code also to show the error, see https://onlinegdb.com/F1_2QHAmK. But your version of the code there is different from this PR.

This, from this PR:
```
    ptr = strstr(pszSource, pszSuffix);
    if (!ptr) {
        return ERROR_TDNF_INVALID_PARAMETER;
```
will also return an error if the suffix is not in the string (`ptr` will be `NULL`).

But that's good.  It's the old version that misbehaves. We don't want to change the string if the suffix doesn't match. And the caller in `client/repo.c` already checks if the suffix matches.